src/Functions/array/range.cpp

```diff
@@ -386,10 +400,27 @@ class FunctionRange : public IFunction
                             "for unsigned/signed integers up to 64 bit", getName());
         }
 
+        auto throwIfNullValue = [&](const ColumnWithTypeAndName & col)
+        {
+            if (!col.type->isNullable())
+                return;
+            const auto & nullable_col = assert_cast<const ColumnNullable &>(*col.column);
```

Comment:
`col.column` can be `Const(Nullable)` here, let's handle this case

Reply:
Fixed (+test)